src/app/select-trajectory/trajectory-card/trajectory-card.component.ts

```diff
@@ -26,6 +26,7 @@ export class TrajectoryCardComponent implements OnInit {
   }
 
   async deleteTrajectory(e: Event) {
+    e.stopPropagation()
```

Comment:
Why do you need that line?

Reply:
I included the delete-icon into the card. Since both elements (delete-icon and card) have their own click-event, both would fire, when the delete-icon is clicked. That line prevents further propagation of the event - therefore the event is only called once.